packages/ui/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {
+  forwardRef,
+  HTMLChakraProps,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  UseNumberInputProps,
+} from '@chakra-ui/react';
+
+type NumberInputSize = 'sm' | 'md' | 'lg';
+
+type NumberInputMode = 'numeric' | 'decimal';
+
+export interface NumberInputProps extends Omit<HTMLChakraProps<'div'>, keyof UseNumberInputProps> {
+  size: NumberInputSize;
+  inputMode: NumberInputMode;
+  disabled: boolean;
```

Comment:
I think it would be easier to use this component if input mode and disabled props were not required. And while destructuring them, set the default value for input mode to decimal and for the disabled prop to false.

Reply:
Also, add short descriptions for the first three props. Check other components how it should be done.